examples/boid_flockers/boid_flockers/server.py

```diff
@@ -5,19 +5,56 @@
 
 
 def boid_draw(agent):
-    return {"Shape": "circle", "r": 2, "Filled": "true", "Color": "Red"}
+    neighbors = agent.model.space.get_neighbors(agent.pos, agent.vision, False)
```

Comment:
This operation is very expensive for all agents, for each step. At which number of agents did you notice any performance degradation?

Reply:
For testing I set 20 FPS. It is fine until 200 agents (which I set as upper limit). After this it slows down kinda linearly until 500, after which it is unbearably slow.

To make the colour process faster I can add the colour logic in the agent class itself. This would make it faster since we anyways need to search neighbourhood there. I can make the change as per your recommendation.